test/e2e/operator_test.go

```diff
@@ -228,3 +233,419 @@ func setImage(deployment *appsv1.Deployment, image string) {
+func lookForStringInPodExec(ns, pod, container string, command []string, expectedString string, timeout time.Duration) error {
+	cmd, err := lookupCmdPath("oc")
+	if err != nil {
+		return err
+	}
+	args := []string{"exec", pod, "-c", container, fmt.Sprintf("--namespace=%v", ns), "--"}
+	args = append(args, command...)
+	if err := lookForString(cmd, args, expectedString, timeout); err != nil {
+		return err
+	}
+	return nil
+}
+
+// lookForStringInPodLog looks for the given string in the log of a specific pod container
+func lookForStringInPodLog(ns, pod, container, expectedString string, timeout time.Duration) error {
+	cmd, err := lookupCmdPath("oc")
+	if err != nil {
+		return err
+	}
+	args := []string{"logs", pod, "-c", container, fmt.Sprintf("--namespace=%v", ns)}
+	if err := lookForString(cmd, args, expectedString, timeout); err != nil {
+		return err
+	}
+	return nil
+}
+
+// lookupCmdPath looks-up the path of cmd returning the full command
+// path or an error.
+func lookupCmdPath(cmdPath string) (string, error) {
+	cmdPath, err := exec.LookPath(cmdPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to find the %s executable in the directories named by the PATH environment variable.", cmdPath)
+	}
+	return cmdPath, nil
+}
+
+// lookForString looks for the given string using cmd and args repeatedly
+// until the timeout is reached or the string is found. Returns an error
+// if the string was not found.
+func lookForString(cmd string, args []string, expectedString string, timeout time.Duration) error {
+	for t := time.Now(); time.Since(t) < timeout; time.Sleep(2 * time.Second) {
+		result, err := runCmd(cmd, args)
+		if err != nil {
+			continue
+		}
+		if strings.Contains(result, expectedString) {
+			return nil
+		}
+	}
+	return fmt.Errorf("failed to find %q", expectedString)
+}
+
+// runCmd runs command cmd with arguments args and returns the output
+// of the command or an error.
+func runCmd(cmd string, args []string) (string, error) {
+	execCmd := exec.Command(cmd, args...)
+	execCmd.Env = append(os.Environ())
+	result, err := execCmd.Output()
+	if err != nil {
+		return "", fmt.Errorf("failed to run command %q with args %q: %v", cmd, args, err)
+	}
+	return string(result), nil
+}
+
+// upstreamConfigMap creates a ConfigMap containing a Corefile
+// configured for the test upstream resolver.
+func upstreamConfigMap(name, ns string) *corev1.ConfigMap {
+	var corefile = `.:5353 {
+    hosts {
+      1.2.3.4 www.foo.com
+    }
+    health
+    errors
+    log
+}
+`
+
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: ns,
+		},
+		Data: map[string]string{"Corefile": corefile},
+	}
+}
+
+// upstreamContainer creates a Container configured for the
+// test upstream resolver.
+func upstreamContainer(container, image string) corev1.Container {
+	dnsPorts := []corev1.ContainerPort{
+		{
+			Name:          "dns",
+			ContainerPort: int32(5353),
+			Protocol:      corev1.Protocol("UDP"),
```

Comment:
I am using `corev1.ProtocolUDP` and `corev1.ProtocolTCP` [here](https://github.com/openshift/cluster-dns-operator/pull/168/files#diff-3fe2c8a818ec73adea75f23c580a9bc0R96-R107).

Reply:
I mean, why not change `corev1.Protocol("UDP")` to `corev1.ProtocolUDP`?  It isn't too important, but you might as well use the named value instead of casting a string.